sklearn/metrics/cluster/unsupervised.py

```diff
@@ -81,6 +82,9 @@ def silhouette_score(X, labels, metric='euclidean', sample_size=None,
            <http://en.wikipedia.org/wiki/Silhouette_(clustering)>`_
 
     """
+    X = np.asarray(X)
```

Comment:
Do it after ``check_X_y`` and you don't have to call ``asarray``. I was hoping there was an option to ``check_X_y`` to enforce this, but it doesn't look like this.